doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,223 @@
+# HTTP Transport Adapter
+
+The Azure SDK for C++ creates and sends HTTP requests using an HTTP Transport Adapter. This adapter provides a general abstraction for an HTTP stack (like libcurl, WinHTTP, etc) by defining the interface it needs to call that anyone can implement to customize which HTTP stack gets used.
+
+## Default HTTP Transport Adapter
+
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl Transport Adapter
+
+<center>
+
+| Supported Platforms  |
+| --- |
+| Linux |
+| Windows |
+| MacOS |
+| [Others](https://daniel.haxx.se/blog/wp-content/uploads/2020/02/slide-operating-systems.jpg) |
```

Comment:
Given the C++ SDK doesn't officially support most of these, I don't think it is worth mentioning this link, even if libcurl itself does work on them.
```suggestion
```